app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+
+    if user&.authenticate(params[:session][:password])
```

Comment:
toán tử này có tác dụng gì em nhỉ `&.`

Reply:
`user&.authenticate `<=> `user && user.authenticate`
`&&` là toán tử logic `AND` cả 2 cùng true.
Kiểm tra `user` có tồn tại và mật khẩu có đúng hay không ạ. Nếu `user nil` thì điều kiện trả về là `nil`.

Reply:
Tìm kiếm chưa đúng keyword rồi em
Em tìm hiểu về `Ruby safe navigation operator` nhé

Reply:
Để gọi `method `của `Object `đó có `nil `hay không hoặc để handle lỗi `NoMethodError: undefined method for nil:NilClass`, để giải quyết thì có method `try()` hay `Safe Navigation Operator (&.)`.cú pháp `&.` chỉ hoạt động khi `object nil` còn trường hợp `false` sẽ đưa ra `exception`.

`&. `giúp code ngắn gọn hơn `try()`. `&.` là cú pháp của Ruby còn phương thức try() không được định nghĩa trong thư viện Ruby core mà là của thư viện Rails.
`&.` giúp debug dễ dàng hơn `(&.)` sẽ đưa ra `exception` nếu 1 phương thức thực sự không tồn tại được gọi, trong khi `try()` luôn luôn trả về `nil`